### PR TITLE
Doc

### PR DESCRIPTION
**Pull request by George-Miao**
This PR adds a centralized document.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,34 @@
 **/target
 /.idea
+
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
```
Comment (PhotonQuantum) [inline .gitignore:7]: dup

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,34 @@
...
+# MacOS temp file
+.DS_Store
+
+book-test
+guide/book
+
+.vscode
+
+# Ignore Jetbrains specific files.
+.idea/
```
Comment (PhotonQuantum) [inline .gitignore:28]: ditto

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,34 @@
...
+**/*.rs.bk
+
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
+
+# MacOS temp file
+.DS_Store
+
+book-test
+guide/book
```
Comment (PhotonQuantum) [inline .gitignore:23]: These entries are non-necessary because `doc/book` is already covered by `book`.

**Review by PhotonQuantum — COMMENTED** [review]
What about deploying the book on vercel for branches and prs, and merge this branch into master, so that it can be available online, allowing us to publish doc of a feature/module even if it's WIP?

**Comment by George-Miao** [comment]
> What about deploying the book on vercel for branches and prs, and merge this branch into master, so that it can be available online, allowing us to publish doc of a feature/module even if it's WIP?

Yes LGTM.

**Pull request merged**